kernel/kernel-keymanager-service/src/main/java/io/mosip/kernel/partnercertservice/util/PartnerCertificateManagerUtil.java

```diff
@@ -274,6 +271,34 @@ private static String buildCertChain(Certificate[] chain) {
         }
     }
 
+    public static String buildCertChainWithPKCS7(Certificate[] chain) {
+        try {
+            CMSSignedDataGenerator generator = new CMSSignedDataGenerator();
+            JcaCertStore jcaStore = new JcaCertStore(Arrays.asList(chain));
+            generator.addCertificates(jcaStore);
+
+            CMSTypedData cmsTypedData = new CMSAbsentContent();
+            CMSSignedData cmsSignedData = generator.generate(cmsTypedData);
+
+            byte[] encodedData = cmsSignedData.getEncoded();
+            String base64Encoded = Base64.getEncoder().encodeToString(encodedData);
+
+            StringBuilder pkcs7Formatted = new StringBuilder();
+            pkcs7Formatted.append("-----BEGIN PKCS7-----\n");
+            pkcs7Formatted.append(base64Encoded.replaceAll("(.{64})", "$1\n"));
+            pkcs7Formatted.append("\n-----END PKCS7-----");
+
+            return pkcs7Formatted.toString();
+        } catch (CertificateEncodingException | CMSException | IOException e) {
+            LOGGER.error(PartnerCertManagerConstants.SESSIONID, PartnerCertManagerConstants.GET_CA_CERT_TRUST,
+                    PartnerCertManagerConstants.PCM_UTIL, "Error generating p7b certificates chain.");
+            throw new PartnerCertManagerException(
+                    PartnerCertManagerErrorConstants.CERTIFICATE_THUMBPRINT_ERROR.getErrorCode(),
```

Comment:
Why is the errorCode "CERTIFICATE_THUMBPRINT_ERROR" ? actually, it should be a conversion error code